Client/EngagementNotificationHelper.swift

```diff
@@ -0,0 +1,82 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+class EngagementNotificationHelper: FeatureFlaggable {
+    struct Constant {
+        #if MOZ_CHANNEL_FENNEC
+        // shorter time interval for development
+        static let timeUntilNotification: UInt64 = UInt64(60 * 2 * 1000) // 2 minutes in milliseconds
+        static let twentyFourHours: UInt64 = timeUntilNotification / 2 // 1 minutes in milliseconds
+        #else
+        static let timeUntilNotification: UInt64 = UInt64(60 * 60 * 48 * 1000) // 48 hours in milliseconds
+        static let twentyFourHours: UInt64 = UInt64(60 * 60 * 24 * 1000) // 24 hours in milliseconds
+        #endif
+        static let notificationId: String = "org.mozilla.ios.engagementNotification"
+    }
+
+    private var notificationManager: NotificationManagerProtocol
+    private var prefs: Prefs?
+    private var firstAppUse: Timestamp? {
+        prefs?.timestampForKey(PrefsKeys.KeyFirstAppUse)
+    }
+    private lazy var featureEnabled: Bool = featureFlags.isFeatureEnabled(.engagementNotificationStatus,
+                                                                          checking: .buildOnly)
+    private var allowedTipsNotifications: Bool {
+        let featureEnabled = featureFlags.isFeatureEnabled(.notificationSettings, checking: .buildOnly)
+        let userPreference = prefs?.boolForKey(PrefsKeys.Notifications.TipsAndFeaturesNotifications) ?? true
```

Comment:
Technically, for cohesion and for note having to use `?? true` we could pass this in through Nimbus, and then check `.userOnly` which would respect Nimbus default if no user setting exists.

Don't necessarily have to change if you don't think it needs to be. I was commenting more with the idea of consolidation of things under nimbus.

Reply:
Good idea. I tried to implement it but it because the toggle is a combination of the user preference and the system notification state it's not easy to do. There is too many edge cases and I don't feel confident it wouldn't cause bugs. So for now I opt to keep it like it is and we can revisit this later.